libs/api_common/windows_platform_common.cpp

```diff
@@ -136,10 +428,726 @@ get_attach_type_windows(const std::string& section)
 _Ret_maybenull_z_ const char*
 get_attach_type_name(_In_ const ebpf_attach_type_t* attach_type)
 {
-    // TODO: (Issue #223) Read the registry to fetch attach types.
-    auto it = windows_section_names.find(*attach_type);
-    if (it != windows_section_names.end())
-        return it->second;
+    for (const ebpf_section_definition_t& t : _windows_section_definitions) {
+        if (IsEqualGUID(*t.attach_type, *attach_type)) {
+            return t.section_prefix;
+        }
+    }
+
+    return nullptr;
+}
+
+static ebpf_result_t
+_load_section_data_information(HKEY section_data_key, _In_ const wchar_t* section_name) noexcept
+{
+    int32_t status;
+    ebpf_result_t result = EBPF_SUCCESS;
+    HKEY section_info_key = nullptr;
+    ebpf_program_type_t* program_type = nullptr;
+    ebpf_attach_type_t* attach_type = nullptr;
+    ebpf_section_definition_t section_definition;
+    bpf_prog_type_t bpf_program_type;
+    bpf_attach_type_t bpf_attach_type;
+    char* section_prefix = nullptr;
+
+    try {
+        status = RegOpenKeyEx(section_data_key, section_name, 0, KEY_READ, &section_info_key);
+        if (status != ERROR_SUCCESS) {
+            // Registry path is not present.
+            result = EBPF_FILE_NOT_FOUND;
+            goto Exit;
+        }
+
+        program_type = (ebpf_program_type_t*)ebpf_allocate(sizeof(ebpf_program_type_t));
+        if (program_type == nullptr) {
+            result = EBPF_NO_MEMORY;
+            goto Exit;
+        }
+
+        attach_type = (ebpf_attach_type_t*)ebpf_allocate(sizeof(ebpf_attach_type_t));
+        if (attach_type == nullptr) {
+            result = EBPF_NO_MEMORY;
+            goto Exit;
+        }
+
+        // Read program type.
+        result = read_registry_value_binary(
```

Comment:
I see a mix of calling Reg APIs directly and invoking such wrappers. If the wrappers are adding value (such as buffer allocation) then use them; ow call Reg APIs directly.

Reply:
Same as above.